capstone/src/main/webapp/scripts/search.js

```diff
@@ -0,0 +1,76 @@
+const searchInput = document.getElementById("name-search");
+const suggestionsPanel = document.getElementById("suggestions");
+
+let names = [
+  { fullname: "Lucy Qu" },
+  { fullname: "Lucy Liu" },
+  { fullname: "Himani Yadav" },
+  { fullname: "Anika Bagga" },
+  { fullname: "Lucille Ball" }
+];
+let currentFocus = -1;
+
+function autocomplete() {
+  searchInput.addEventListener("keyup", filter);
+  searchInput.addEventListener("keyup", checkKey);
+}
+
+function filter() {
+  const input = searchInput.value.toLowerCase();
+  suggestionsPanel.innerHTML = "";
+  const suggestions = names.filter(function(name) {
+    return name.fullname.toLowerCase().startsWith(input);
+  });
+  suggestions.forEach(suggested => appendSuggestion(suggested));
+  if (input === "") {
+    suggestionsPanel.innerHTML = "";
+  }
```

Comment:
Instead of clearing the suggestionsPanel if the input is empty at the end of the function, couldn't you just only add the suggestions if the input is not empty?

Reply:
This is here because if I type a name it'll pull up results but then if I backspace all the way to nothing, the last posted suggestions are still going to be there, so I do this check to clear it. I tried doing what you suggested but the names still stay there

Reply:
edited in new pr